crates/jstz_node/src/services/logs/db.rs

```diff
@@ -0,0 +1,137 @@
+        let manager = SqliteConnectionManager::file(db_path);
+        let pool = SqliteConnectionPool::new(manager)?;
+
+        Self::create_table(pool.clone()).await?;
+
+        Ok(Db { pool })
+    }
+
+    async fn create_table(pool: Pool<SqliteConnectionManager>) -> Result<()> {
+        let connection = Self::get_connection_from_pool(pool).await?;
+
+        connection.execute_batch(include_str!("./create_db.sql"))?;
+
+        Ok(())
+    }
+
+    pub async fn connection(&self) -> Result<SqliteConnection> {
+        Self::get_connection_from_pool(self.pool.clone()).await
+    }
+
+    async fn get_connection_from_pool(
+        pool: SqliteConnectionPool,
+    ) -> Result<SqliteConnection> {
+        block(move || pool.get())
+            .await
+            .map_err(|e| {
+                anyhow!("Failed to get connection from pool: {}", e.to_string())
+            })?
+            .map_err(|e| anyhow!("Failed to get connection from pool: {}", e.to_string()))
+    }
+
+    // On success, returns the number of rows that were changed/inserted.
+    pub(super) async fn flush(&self, line: &Line) -> Result<()> {
+        let connection = self.connection().await?;
+        match line {
+            Line::Request(RequestEvent::Start {
+                request_id,
+                contract_address,
+            }) => connection.execute(
+                "INSERT INTO request (id, function_address) VALUES (?1, ?2)",
+                (request_id, contract_address.to_string()),
+            )?,
+            Line::Js(LogRecord {
+                request_id,
+                contract_address,
+                level,
+                text,
+            }) => connection.execute(
+                "INSERT INTO log (level, content, function_address, request_id) VALUES (?1, ?2, ?3, ?4)",
+                (
+                    level.to_string(),
+                    text,
+                    contract_address.to_string(),
+                    request_id
+                ),
+            )?,
+            // TODO: Update the request row with more fields.
+            Line::Request(_) => 0,
+        };
+
+        Ok(())
+    }
+
+    pub async fn logs_by_address(
+        &self,
+        function_address: Address,
+        limit: usize,
+        offset: usize,
+    ) -> QueryResponseResult {
+        let conn = self.connection().await?;
+
+        let stmt = conn
+            .prepare("SELECT * FROM log WHERE function_address = ? LIMIT ? OFFSET ?")?;
+
+        Self::collect_logs(stmt, params![function_address.to_string(), limit, offset])
+    }
+
+    pub async fn logs_by_address_and_request_id(
+        &self,
+        function_address: Address,
+        request_id: String,
+    ) -> QueryResponseResult {
+        let conn = self.connection().await?;
+
+        let stmt = conn
+            .prepare("SELECT * FROM log WHERE function_address= ? AND request_id= ?")?;
+
+        Self::collect_logs(stmt, [function_address.to_string(), request_id])
```

Comment:
Missing `params!` macro

Reply:
when the element is of the same type we can omit the params1 :)